### ISSUE TITLE
[Question] Is this project dead/abandoned?

### ISSUE THREAD

**Issue by netbull** [suncat2000/MobileDetectBundle#138]
First of all, thank you for the bundle! It's quite handy in many ways.

But as a cool project is quite unmaintained lately.. last PR merge is almost 2 years ago. The most vital PRs are for compatibility with Symfony 4.4 and 5 which is out now.

**Comment by DennisOluoch** [comment]
i am also concerned

**Comment by pwnyprod** [comment]
same here, will at least the dependencies updated for symfony5?

**Comment by DennisOluoch** [comment]
I don't think so...you will need to look out for the forks and replace this bundles with your preferred fork to get an update

**Comment by pwnyprod** [comment]
I tried start to refactor (huge pain for 2 hours), then i go over to forget this bundle and write a twig extension for my application and easily readout the server params (same like mobiledetection dependency does for mobile detection) so for me this is huge overhead and from the codebase it consume more time to refactor this. Maybe its easier to write a new Bundle.

**Comment by netbull** [comment]
@pwnyprod I don't know exactly what you refactored, but I forked the bundle as well and it took an hour or so and it works just fine with Symfony 4.4.x `netbull/mobile-detect-bundle`

**Comment by pwnyprod** [comment]
@netbull i use Symfony 5.0.x

**Comment by pwnyprod** [comment]
So lets question if @suncat2000 would give you the official branch.

**Comment by evertharmeling** [comment]
Seems like https://github.com/tattali/MobileDetectBundle is a good updated alternative, based on this repo!

**Comment by truckee** [comment]
Note that [https://github.com/tattali/MobileDetectBundle](https://github.com/tattali/MobileDetectBundle) cannot be installed with `composer require tattali/MobileDetectBundle`:

>  [InvalidArgumentException]
  Package tattali/MobileDetectBundle not found

**Comment by pwnyprod** [comment]
because its not on packagist, you have to add a vcs link in you package.json but anyway this is getting more and more outdated. A good alternative is https://github.com/Gregwar/ImageBundle

**Comment by netbull** [comment]
> because its not on packagist, you have to add a vcs link in you package.json but anyway this is getting more and more outdated. A good alternative is https://github.com/Gregwar/ImageBundle

How image bundle is related with mobile detect?

**Comment by pwnyprod** [comment]
oh nvm :D too many projects, it was the wrong thread 🦩

**Comment by evertharmeling** [comment]
> Note that https://github.com/tattali/MobileDetectBundle cannot be installed with `composer require tattali/MobileDetectBundle`:
> 
> > [InvalidArgumentException]
> > Package tattali/MobileDetectBundle not found

You should install it with `composer require tattali/mobile-detect-bundle` (see https://packagist.org/packages/tattali/mobile-detect-bundle)

**Comment by netbull** [comment]
`netbull/mobile-detect-bundle` now can be installed via composer on Symfony 5 project

**Comment by truckee** [comment]
Bundle Installs just fine. Yaml file needs to be resurrected from docs. After modifying `host: ...` in `config/packages/mobile-direct.yaml` I get

>Attempted to load class "MobileDetectExtension" from namespace "MobileDetectBundle\Twig\Extension".
Did you forget a "use" statement for e.g. "SunCat\MobileDetectBundle\DependencyInjection\MobileDetectExtension" or "SunCat\MobileDetectBundle\Twig\Extension\MobileDetectExtension"?

**Comment by lolop93** [comment]
@netbull  Service "mobile_detect.mobile_detector" not found: even though it exists in the app's container....

![Captura de pantalla (14)](https://user-images.githubusercontent.com/20732720/129023536-af9a4ec2-d8b2-42b3-a576-b0524c3132c9.png)

it give me this error 


and this is my code 
![Captura de pantalla (15)](https://user-images.githubusercontent.com/20732720/129023611-35c35e9f-bcbf-4f15-bff8-6a3bc89703a7.png)


symfony 5 and php 8

**Comment by netbull** [comment]
@lolop93 you will need to inject the MobileDetect just like you inject the repositories.

**Comment by lolop93** [comment]
> @lolop93 you will need to inject the MobileDetect just like you inject the repositories.

@netbull i dont know how....

![image](https://user-images.githubusercontent.com/20732720/129030181-58f31779-42f9-44fc-ac97-b7961b54b5da.png)

this is i have

still dont work

and this neither dont work use SunCat\MobileDetectBundle\MobileDetectBundle;

![image](https://user-images.githubusercontent.com/20732720/129030867-9ece5cfb-9b53-45bb-a096-a5a4a938b9b7.png)

i do the inyection but give me this error
![image](https://user-images.githubusercontent.com/20732720/129030954-35f0d805-23fb-49b7-9736-241c087a6591.png)

**Comment by truckee** [comment]
@lolop93 FWIW, it is possible to gain the effects of the bundle using just the [Twig Helper](https://github.com/suncat2000/MobileDetectBundle/blob/master/Resources/doc/index.md#twig-helper). No need to inject it. For example

```
{# base.html.twig #}
...
            {% if is_mobile_view() %}
                {% block nav_mobile %}
                    {% include 'Default/mobile_menu.html.twig' %}
                    {% include 'Default/top_menu.html.twig' %}
                {% endblock nav_mobile %}
            {% endif %}
```

**Comment by lolop93** [comment]
> @truckee @lolop93 FWIW, it is possible to gain the effects of the bundle using just the [Twig Helper](https://github.com/suncat2000/MobileDetectBundle/blob/master/Resources/doc/index.md#twig-helper). No need to inject it. For example
> 
> ```
> {# base.html.twig #}
> ...
>             {% if is_mobile_view() %}
>                 {% block nav_mobile %}
>                     {% include 'Default/mobile_menu.html.twig' %}
>                     {% include 'Default/top_menu.html.twig' %}
>                 {% endblock nav_mobile %}
>             {% endif %}
> ```

yes i know but i prefer inject it on the controller for a better performance and readability

The twig helper works perfectly :)
